packages/vite/src/node/plugins/html.ts

```diff
@@ -323,6 +323,22 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
             config,
             publicToRelative,
           )
+        const nodeStartWithLeadingWhitespace = (
+          node: DefaultTreeAdapterMap['node'],
+        ) => {
+          if (node.sourceCodeLocation!.startOffset == 0)
+            return node.sourceCodeLocation!.startOffset
+          const lineStartOffset =
+            node.sourceCodeLocation!.startOffset -
+            node.sourceCodeLocation!.startCol
```

Comment:
I think it'd nice to have a comment that this variable points the line break. Or change `node.sourceCodeLocation!.startCol` to `node.sourceCodeLocation!.startCol - 1`, so that it points the 0-indexed line start.

Reply:
Sure

> Or change `node.sourceCodeLocation!.startCol` to `node.sourceCodeLocation!.startCol - 1`, so that it points the 0-indexed line start.

Not quite sure I follow, columns are not 0-indexed. Add a `- 1` and you're pointing to the previous line.

Reply:
I mean I think `lineStartOffset` points the line break in the previous line.
```html
<!-- -->
<script></script>
```
In this example HTML, the values are:
- `node.sourceCodeLocation!.startOffset` (0-indexed): 1
- `node.sourceCodeLocation!.startCol` (1-indexed): 1
- `lineStartOffset` (0-indexed): 0 (the line break in the previous line)

If `lineStartOffset` is changed like below, `lineStartOffset` will point the start of the line.
```js
const lineStartOffset =
  node.sourceCodeLocation!.startOffset -
  (node.sourceCodeLocation!.startCol - 1) // convert to 0-index
```

Reply:
No, `node.sourceCodeLocation!.startOffset - (node.sourceCodeLocation!.startCol - 1)` will end up selecting (and removing) the last character (not the newline) of the previous line, likely an end tag unless the user has trailing whitespace.

Reply:
Sorry, deleted my first response as I misread.

I'm not sure what the point is in the change? Sure, you could point to the newline and change the regex, but that doesn't offer anything that I'm aware of?

Reply:
What I was suggesting is that if this variable (`lineStartOffset`) is pointing to a newline, then either

- Clarify in the comment that it is pointing to the line break in the previous line
  - I think the current comment (`Gets the offset for the start of the line that the node is on`) is missing the information that it's pointing to the line break.
- Change the code to point to the beginning of the line instead of the line break in the previous line, so that the variable name is correct

Reply:
I'd consider the newline part of the "line", hence the name -- when you insert a new tag into an HTML (especially in the head) you'd always append a newline to the end of the previous. The newline at the end of one element is in all practicality the line start for the next element.

But if you want it changed I can do that I suppose.

Reply:
Ah, I see. That makes sense as well. Thank you!